adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -377,6 +377,36 @@ private void processRemoteOpen(final ProtonConnection con) {
             TracingHelper.setDeviceTags(span, authenticatedDevice.getTenantId(), authenticatedDevice.getDeviceId());
         }
 
+        checkConnectionLimitForAdapter()
```

Comment:
My first intuition here would have been to keep the order of the checks and do the (usually less expensive) tenant and device checks first. But, thinking more about it, I agree that doing the connection limit check first seems more intuitive and logical. In any case, that is also how it is done in the MQTT adapter.

Reply:
actually, the adapter connection limit check is the cheapest one of all as it doesn't require any remote service invocation at all but simply checks an integer that is managed within the Metrics implementation class ...

Reply:
right, I confused it with the message limit check.